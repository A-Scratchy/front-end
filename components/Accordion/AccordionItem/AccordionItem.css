.accordionSingleHidden {
  display: none;
}

.accordionSingleContent {
  margin-top: 5px;
  max-height: 0;
  opacity: 0;
  transition: all 0.4s ease;
  position: relative;
  max-width: 100%;
}

.accordionSingleHidden:not(:checked) ~ .accordionSingleContent {
  height: 0;
  margin: 0;
}

.accordionSingleHidden:checked ~ .accordionSingleContent {
  max-height: 1000px;
  opacity: 1;
  margin-top: 14px;
}

.accordionSingle {
  border: 1px solid #c5c5c5;
  padding: 20px;
  border-radius: 5px;
  margin-bottom: 10px;
  width: 100%;
}

.accordionSingleTitle {
  font-weight: 600;
  text-transform: uppercase;
  cursor: pointer;
  position: relative;
}

.accordionSingleTitle:hover,
.accordionSingleTitle:focus {
  text-decoration: underline;
}

.accordionSingleTitle::before {
  display: inline-block;
  margin-right: 10px;
  color: var(--primary);
  content: "+";
}

.accordionSingleHidden:checked ~ .accordionSingleTitle::after {
  content: "-";
}

@media screen and (--medium-viewport) {
  .accordionSingle {
    max-width: 270px;
  }
}

@media screen and (--small-viewport) {
  .accordionSingle {
    max-width: 200px;
  }
}
