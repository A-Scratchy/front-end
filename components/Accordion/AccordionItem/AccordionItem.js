import React from 'react';
import { string, node } from 'prop-types';
import styles from './AccordionItem.css';

AccordionItem.propTypes = {
  title: string.isRequired,
  content: node.isRequired,
};

function AccordionItem({ title, content }) {
  // Create an HTML-safe ID for input query selecting
  const titleID = title.replace(/[.,/#!?@#$%^&*;:{}=\-_'`~()]/g, '').replace(/\s{2,}/g, ' ');

  return (
    <div className={styles.accordionSingle}>
      <input className={styles.accordionSingleHidden} type="checkbox" id={titleID} />

      <label className={styles.accordionSingleQuestion} htmlFor={titleID}>
        {title}
      </label>

      <p className={styles.accordionSingleAnswer}>{content}</p>
    </div>
  );
}

export default AccordionItem;
