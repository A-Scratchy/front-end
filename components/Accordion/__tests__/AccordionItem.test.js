import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';

import AccordionItem from '../AccordionItem/AccordionItem';

describe('AccordionItem', () => {
  it('should render with required props', () => {
    createSnapshotTest(<AccordionItem title="How do I test things?" content={<>Like so!</>} />);
  });
});
