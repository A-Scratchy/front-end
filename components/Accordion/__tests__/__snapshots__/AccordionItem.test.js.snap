// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionItem should render with required props 1`] = `
<div
  className="accordion"
>
  <input
    className="accordionHidden"
    id="How do I test things"
    type="checkbox"
  />
  <label
    className="accordionTitle"
    htmlFor="How do I test things"
  >
    How do I test things?
  </label>
  <p
    className="accordionContent"
  >
    Like so!
  </p>
</div>
`;
